lib/blank_video_generator.ex

```diff
@@ -0,0 +1,92 @@
+defmodule Membrane.BlankVideoGenerator do
+  @moduledoc """
+  Element responsible for generating black screen as raw video.
+  """
+
+  use Membrane.Source
+
+  alias Membrane.{Buffer, Time}
+  alias Membrane.Caps.Matcher
+  alias Membrane.Caps.Video.Raw
+
+  @supported_caps {Raw, format: Matcher.one_of([:I420, :I422]), aligned: true}
+
+  def_options duration: [
+                type: :integer,
+                spec: Time.t(),
+                description: "Duration of the output"
+              ],
+              caps: [
+                type: :struct,
+                spec: Raw.t(),
+                description: "Video format of the output"
+              ]
+
+  def_output_pad :output,
+    caps: @supported_caps,
+    mode: :pull,
+    availability: :always
+
+  @impl true
+  def handle_init(opts) do
+    inspect(opts)
+
+    if Matcher.match?(@supported_caps, opts.caps) do
```

Comment:
I think you can also check if either width (do I remember correctly?) or width and height can be divided by 2 for i420 and i422 respectivly.